### ISSUE TITLE
CoreCompile does not consider imported rulesets as inputs

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#25041]
The `CoreCompile` target considers the main ruleset file for a compilation as an input, so changes to the file will cause the build output to change:

https://github.com/dotnet/roslyn/blob/aa683ac4ec4c6ce746776a2a316309fe6a7d1740/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L29

However, ruleset files are allowed to import other ruleset files:

https://github.com/dotnet/roslyn/blob/877537101920994db0cbb7fd3bcf25bb391860eb/build/Rulesets/NonShippingProject_BuildRules.ruleset#L3

These imported files are not tracked by the `ResolvedCodeAnalysisRuleSet` property, and changes to them do not trigger a build.

**Comment by jcouv** [comment]
@agocke Feels like we should keep this issue in 16.3. May want to load-balance it though.

**Comment by Youssef1313** [comment]
@mavasani @sharwell Do we care about ruleset issues anymore?

**Comment by mavasani** [comment]
> @mavasani @sharwell Do we care about ruleset issues anymore?

I don’t think we want to do anything here unless there is strong customer requirement for it
